watcher/watcher.go

```diff
@@ -104,29 +113,48 @@ func CreateWatcher(watchable Watchable, period time.Duration, maxFailures int, t
 		prometheus.CounterOpts{
 			Name: prefix + "dropped_message_count",
 			Help: "Number of messages that were dropped"})
+	totalRoundTripsMetric := prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: prefix + "total_round_trips",
+			Help: "Number of sucessful round trips"})
+	averageRoundTripTimeMetric := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: prefix + "average_round_trip_time",
+			Help: "Average Round Trip Time in seconds"})
+	runningAverageRoundTripTimeMetric := prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: prefix + "running_average_round_trip_time",
+			Help: fmt.Sprintf("Running Average Round Trip Time in seconds for last %d messages",
+				RUNNING_AVERAGE_SLOTS)})
 	prometheus.MustRegister(writeFailuresMetric)
 	prometheus.MustRegister(readFailuresMetric)
 	prometheus.MustRegister(statusMetric)
 	prometheus.MustRegister(droppedMessagesMetric)
 	prometheus.MustRegister(minRoundTripTimeMetric)
 	prometheus.MustRegister(maxRoundTripTimeMetric)
+	prometheus.MustRegister(averageRoundTripTimeMetric)
```

Comment:
Register this in the same order like you initialized them in lines 116-128.

Reply:
Good idea. Done